webapp/lib/app/nook/view.dart

```diff
@@ -413,34 +426,86 @@ class ConversationPanelView with AutomaticSuggestionIndicator {
   }
 }
 
+class DateSeparatorView {
+  DateTime _dateTime;
+  String _formattedDateString;
+  String get formattedDateString => _formattedDateString;
+
+  DivElement renderElement;
+
+  DateSeparatorView(this._dateTime) {
+    renderElement = DivElement()..className = 'messages-date-separator__wrapper';
+    _formattedDateString = _getDaysAgoSinceToday(this._dateTime);
+    var dateElement = SpanElement()
+      ..className = 'messages-date-separator'
+      ..innerText = _formattedDateString;
+    renderElement.append(dateElement);
+  }
+
+  void hide() {
+    renderElement.setAttribute('hidden', 'true');
+  }
+
+  void show() {
+    renderElement.removeAttribute('hidden');
+  }
+
+  String _getDaysAgoSinceToday(DateTime dateTime) {
+    final today = DateTime.now();
+    final difference = today.difference(dateTime);
```

Comment:
To make the difference in days work as expected, we need to first remove the number of hours/minutes etc., from both datetimes, otherwise we would get strange behaviour like all messages sent yesterday evening will appear as sent today, if it's only morning (ie. difference between datetimes is < 24h, but across 2 weekdays).